File: translations/codenet/mistral_commenetd/C++2P/granite/Code/s447383972.py

# Python code

# Function to read input
def read():
    return int(input())

# Function to write output
def write(x):
    print(x, end=" ")

# Function to write output on new line
def writeln(x):
    print(x)

# Main function
if __name__ == "__main__":
    # Reading input
    n = read()

    # Reading array 'a'
    a = [read() for _ in range(n)]

    # Sorting array 'a' in ascending order
    a.sort()

    # Finding duplicate elements and storing them in 'q'
    q = []
    for i in range(n):
        if i < n-1 and a[i] == a[i+1]:
            q.append(a[i])
            i += 1

    # Outputting the result
    writeln(q[-1] * q[-2])

# End of Python code

File: translations/codenet/mistral_commenetd/C++2J/gpt-4o/Code/s447383972.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    static final int N = 200010;
    static long[] a = new long[N];
    static long[] q = new long[N];
    static long tot = 0;

    // Function to read input
    static long read() {
        Scanner scanner = new Scanner(System.in);
        return scanner.nextLong();
    }

    // Function to write output
    static void writeln(long x) {
        System.out.println(x);
    }

    public static void main(String[] args) {
        // Reading input
        long n = read();

        // Reading array 'a'
        for (long i = 1; i <= n; i++) {
            a[(int) i] = read();
        }

        // Sorting array 'a' in ascending order
        Arrays.sort(a, 1, (int) (n + 1));

        // Finding duplicate elements and storing them in 'q'
        for (long i = 1; i <= n; i++) {
            if (a[(int) i] == a[(int) (i + 1)]) {
                q[(int) ++tot] = a[(int) i];
                i++; // Incrementing index 'i'
            }
        }

        // Outputting the result
        writeln(q[(int) tot] * q[(int) (tot - 1)]);
    }
}
// <END-OF-CODE>
